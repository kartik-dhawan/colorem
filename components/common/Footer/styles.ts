export const styles = {
  footerBoxWrapper: {
    backgroundColor: "#111",
    borderTop: "1px solid #333",
    // background: "linear-gradient(180deg, #111 40%, #222 95%)",
    padding: "2rem 0rem",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  footerBoxIcon: {
    margin: "5px 15px",
    color: "#d9d9d9",
    fontSize: {
      xs: "45px",
      sm: "53px",
    },
    cursor: "pointer",
  },
  footerBoxBody: {
    textAlign: "center",
    fontSize: {
      xs: "18px",
      sm: "19px",
    },
    padding: {
      xs: "0rem 2rem",
      sm: "0rem 4rem",
    },
    fontWeight: 300,
    fontFamily: "Nunito Sans",
    color: "#ebe1e1", // text_third
    "& > a": {
      color: "#0071f3f5",
      transition: "100ms all ease-in-out",
      "&:hover": {
        color: "#3688e6f5", // next_blue - just lighter
      },
    },
  },
  footerBoxReferenceWrapper: {
    display: "flex",
    flexWrap: "wrap",
    margin: "30px 0px",
    justifyContent: "center",
  },
  footerBoxReference: {
    "& > a": {
      color: "#c7b4b4", // text_primary_dull
      fontSize: "17px",
      padding: {
        xs: "0px 20px",
        sm: "0px 30px",
      },
      "&:hover": {
        color: "#d9d9d9", // text_primary
      },
    },
  },
}
