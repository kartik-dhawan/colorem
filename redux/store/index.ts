import { configureStore } from "@reduxjs/toolkit"
import countSlice from "../slices/countSlice"

const store = configureStore({
  reducer: {
    countSlice,
  },
})

export default store
