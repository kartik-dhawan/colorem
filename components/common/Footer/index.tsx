import { Box, Typography } from "@mui/material"
import GitHubIcon from "@mui/icons-material/GitHub"
import InstagramIcon from "@mui/icons-material/Instagram"
import LinkedInIcon from "@mui/icons-material/LinkedIn"
import Link from "next/link"
import { styles } from "./styles"
import { externalLinks } from "../../../utils/constants"
import { useSelector } from "react-redux"
import { RootType } from "../../../redux/constants/stateTypes"

const Footer = () => {
  const frid = "footerBox"
  let mapId = 0

  const content = useSelector((state: RootType) => state.contentSlice.data)
  return (
    <Box
      className={frid + "Wrapper"}
      id={frid + "Wrapper"}
      sx={styles.footerBoxWrapper}
    >
      <Box
        className={frid + "IconsWrapper"}
        id={frid + "IconsWrapper"}
        sx={{
          padding: "20px 0px",
        }}
      >
        <Link href={externalLinks.GITHUB} target="_blank">
          <GitHubIcon sx={styles.footerBoxIcon} />
        </Link>
        <Link href={externalLinks.INSTAGRAM} target="_blank">
          <InstagramIcon sx={styles.footerBoxIcon} />
        </Link>
        <Link href={externalLinks.LINKEDIN} target="_blank">
          <LinkedInIcon sx={styles.footerBoxIcon} />
        </Link>
      </Box>
      <Typography
        variant="body1"
        className={frid + "Body"}
        id={frid + "Body"}
        sx={styles.footerBoxBody}
      >
        {content.footerDevelopedByText}{" "}
        <Link href="/about">
          <u>{content.developedBy}</u>
        </Link>
        <br></br>
        {content.footerReferenceText}
      </Typography>
      <Box
        className={frid + "ReferenceWrapper"}
        id={frid + "ReferenceWrapper"}
        sx={styles.footerBoxReferenceWrapper}
      >
        {content.references?.map((item: any) => {
          mapId++
          return (
            <Box
              className={frid + "Reference"}
              id={frid + "Reference_" + mapId}
              sx={styles.footerBoxReference}
              key={mapId}
            >
              <Link href={item.source} target="_blank" className={"refLink"}>
                {item.title}
              </Link>
              {mapId < 3 && <span>|</span>}
            </Box>
          )
        })}
      </Box>
    </Box>
  )
}

export default Footer
