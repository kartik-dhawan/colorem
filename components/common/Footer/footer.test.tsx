import { Provider } from "react-redux"
import renderer from "react-test-renderer"
import "@testing-library/jest-dom"
import store from "../../../redux/store"
import Footer from "."

describe("Footer section Component", () => {
  it("-- should match snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Footer />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
