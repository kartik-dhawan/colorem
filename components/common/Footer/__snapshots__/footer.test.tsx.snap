// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer section Component -- should match snapshot 1`] = `
<div
  className="footerBoxWrapper MuiBox-root css-1kirzbg"
  id="footerBoxWrapper"
>
  <div
    className="footerBoxIconsWrapper MuiBox-root css-1mh511h"
    id="footerBoxIconsWrapper"
  >
    <a
      href="https://github.com/kartik-dhawan"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-10egbfp-MuiSvgIcon-root"
        data-testid="GitHubIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
        />
      </svg>
    </a>
    <a
      href="https://instagram.com/notkartikk"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-10egbfp-MuiSvgIcon-root"
        data-testid="InstagramIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
        />
      </svg>
    </a>
    <a
      href="https://www.linkedin.com/in/kartik-dhawan-13585b211/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-10egbfp-MuiSvgIcon-root"
        data-testid="LinkedInIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
        />
      </svg>
    </a>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body1 footerBoxBody css-xtk4il-MuiTypography-root"
    id="footerBoxBody"
  >
     
    <a
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <u />
    </a>
    <br />
  </p>
  <div
    className="footerBoxReferenceWrapper MuiBox-root css-1s51bno"
    id="footerBoxReferenceWrapper"
  />
</div>
`;
